job-profile-wrangler/src/main/java/org/folio/graph/GraphReader.java

```diff
@@ -0,0 +1,219 @@
+import org.jgrapht.Graph;
+import org.jgrapht.graph.SimpleDirectedGraph;
+import org.jgrapht.nio.Attribute;
+import org.jgrapht.nio.dot.DOTImporter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.folio.graph.GraphWriter.DOT_FILE_PATTERN;
+
+/**
+ * The GraphReader class provides methods to read and search graphs from DOT files.
+ * It uses the JGraphT library to represent and manipulate the graphs.
+ */
+public class GraphReader {
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static final DOTImporter<Profile, RegularEdge> DOT_IMPORTER = new DOTImporter<>();
+
+  static {
+    // Configure the DOTImporter to create vertices based on the "name" attribute
+    DOT_IMPORTER.setVertexWithAttributesFactory((id, attrs) -> {
+      Map<String, String> normalAttrs = normalizeAttributes(attrs);
+      String name = normalAttrs.get("name");
+      switch (name) {
+        case "Job Profile" -> {
+          return JobProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Match Profile" -> {
+          return MatchProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Action Profile" -> {
+          return ActionProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Mapping Profile" -> {
+          return MappingProfileNode.fromAttributes(id, normalAttrs);
+        }
+        default -> {
+          LOGGER.error("Unrecognized profile: '{}'", normalAttrs);
+          return null;
+        }
+      }
+    });
+
+    // Configure the DOTImporter to create edges based on the "label" attribute
+    DOT_IMPORTER.setEdgeWithAttributesFactory((attrs -> attrs.entrySet()
+      .stream()
+      .filter(entry -> entry.getKey().equals("label"))
+      .findFirst()
+      .map(entry -> {
+        Attribute attr = entry.getValue();
+        if (MatchRelationshipEdge.getLabelValue().equals(attr.getValue())) {
+          return new MatchRelationshipEdge();
+        } else if (NonMatchRelationshipEdge.getLabelValue().equals(attr.getValue())) {
+          return new NonMatchRelationshipEdge();
+        }
+        return null;
+      })
+      .orElse(new RegularEdge())
+    ));
+  }
+
+  private GraphReader() {}
+
+  /**
+   * Reads a graph from a DOT file specified by the repository path and ID.
+   *
+   * @param repoPath The path to the repository containing the DOT files.
+   * @param id       The ID of the graph to read.
+   * @return The read graph.
+   */
+  public static Graph<Profile, RegularEdge> read(String repoPath, int id) {
+    Path filePath = Paths.get(repoPath, GraphWriter.genGraphFileName(id));
+    Graph<Profile, RegularEdge> g = new SimpleDirectedGraph<>(RegularEdge.class);
```

Comment:
@okolawole-ebsco  I believe that variable "g" has invalid name(sonar can fail).Let's rename it to "simpleGraph"?

Reply:
updated.